app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -4,32 +4,68 @@ import androidx.appcompat.app.AppCompatActivity
 import android.view.View
 import android.widget.Button
+import android.widget.ImageButton
+import android.widget.TextView
 import android.widget.Toast
 
 class MainActivity : AppCompatActivity() {
     private lateinit var trueButton: Button
     private lateinit var falseButton: Button
+    private lateinit var prevButton: ImageButton
+    private lateinit var nextButton: ImageButton
+    private lateinit var questionTextView: TextView
 
+    private val questionBank = listOf(
+            Question(R.string.question_australia, true),
+            Question(R.string.question_oceans, true),
+            Question(R.string.question_mideast, false),
+            Question(R.string.question_africa, false),
+            Question(R.string.question_americas, true),
+            Question(R.string.question_asia, true))
+    private var currentIndex = 0
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
         trueButton = findViewById(R.id.true_button)
         falseButton = findViewById(R.id.false_button)
-
+        prevButton = findViewById(R.id.prev_button)
+        nextButton = findViewById(R.id.next_button)
+        questionTextView = findViewById(R.id.question_text_view)
         trueButton.setOnClickListener { view: View ->
-            Toast.makeText(
-                    this,
-                    R.string.correct_toast,
-                    Toast.LENGTH_SHORT)
-                    .show()
+            checkAnswer(true)
         }
         falseButton.setOnClickListener { view: View ->
-            Toast.makeText(
-                    this,
-                    R.string.incorrect_toast,
-                    Toast.LENGTH_SHORT)
-                    .show()
+            checkAnswer(false)
+        }
+        nextButton.setOnClickListener {
+            currentIndex = (currentIndex + 1) % questionBank.size
+            updateQuestion()
+        }
+        prevButton.setOnClickListener {
+            if (currentIndex - 1 < 0) {
+                currentIndex = questionBank.size - 1
+            } else {
+                currentIndex = (currentIndex - 1) % questionBank.size }
+            updateQuestion()
+        }
+        questionTextView.setOnClickListener { view: View ->
+            currentIndex = (currentIndex + 1) % questionBank.size
+            updateQuestion()
+        }
+        updateQuestion()
+    }
+    private fun updateQuestion() {
+        val questionTextResId = questionBank[currentIndex].textResId
+        questionTextView.setText(questionTextResId)
+    }
+    private fun checkAnswer(userAnswer: Boolean) {
+        val correctAnswer = questionBank[currentIndex].answer
+        val messageResId = if (userAnswer == correctAnswer) {
+            R.string.correct_toast
+        } else {
+            R.string.incorrect_toast
         }
+        Toast.makeText(this, messageResId, Toast.LENGTH_SHORT)
+                .show()
     }
 }
```

Comment:
we add new lines at the end of the file that could sound wired but you can read the explanation [here](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file)